src/seed.rs

```diff
@@ -27,32 +27,23 @@ pub struct Seed {
 
 impl Rand for Seed {
     fn rand<R: Rng>(rng: &mut R) -> Seed {
-        let mut seq: Vec<usize> = (0..TABLE_SIZE).collect();
-        for i in (0..TABLE_SIZE) {
-            let mut swap_i: usize = rng.gen();
-            swap_i = swap_i % TABLE_SIZE;
-            let swap = seq[swap_i];
-            seq[swap_i] = seq[i];
-            seq[i] = swap;
-        }
+        let mut seq = [0; TABLE_SIZE];
+        for (i, x) in seq.iter_mut().enumerate() { *x = i }
+        rng.shuffle(&mut seq);
```

Comment:
This is equivalent. The switch to the shuffle is definitely an improvement over the for loop, but it seems to me that:

    let mut seq: Vec<usize> = (0..TABLE_SIZE).collect();

is cleaner than

    let mut seq = [0; TABLE_SIZE];
    for (i, x) in seq.iter_mut().enumerate() { *x = i }

It's shorter, easier to read (in my opinion), and only initializes the data once, instead of filling it with 0s and then the real values.